GitHub Actions / YAML Section Consistency (#228)
carpentries-incubator/python-intermediate-development: _episodes/23-continuous-integration-automated-testing.md

[YAML](https://www.commonwl.org/user_guide/yaml/)
(a recursive acronym which stands for "YAML Ain't Markup Language")
is a language designed to be human readable.

> The four basic things you need to know about YAML to get started with GitHub Actions are

Reviewer: ```suggestion
A few basic things you need to know about YAML to get started with GitHub Actions are
```

Reviewer: Perhaps we should not count the things as clearly they change :-).

Reviewer 2: Yes, I think this is a good amendment!

~~~
first_scaled_by:

> - Hans Meyer

Reviewer: ```suggestion
- Hans Meyer
```

Reviewer: We have checked with online YAML linting tool https://www.yamllint.com/ and it prefers the indentation as was originally done in the material. I do not mind either way as both are correct - it's just the readability that matters in this case. Maybe @steve-crouch can weight in on this one?

Reviewer 2: @aleks On balance I'd say go with two spaces of indentation as it is, since it appears to be the normal convention in this case (well, either 2 or four, but some claim 4 spaces can decrease readability in the event of long lines and wrapping)

Author: Thanks for both your thoughts - I guess I'm a bit conflicted!

Making it consistent with yamllint makes sense, and I agree this format is easier to read, but having it be different from the GitHub actions later on (when that is the purpose of the YAML introduction) feels like it could trip someone up. 

We could reformat the GH Actions, but that _also_ seems like a bad idea, as then it will be different from the templates GitHub provide.

I'm happy to drop this commit if that's what everyone else thinks is best 🙂

Reviewer: OK, let's then go with the formatting used by GH Actions then - that does make sense and will be used later on in the course and both formatting are correct and we are not breaking anything.

Author: Great - from my perspective please merge whenever :)
